src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -100,57 +100,177 @@ public class ChangeManagement implements Serializable {
         }
     }
 
-    void uploadFileToTransportRequest(BackendType type, String changeId, String transportRequestId, String applicationId, String filePath, String endpoint, String credentialsId, String cmclientOpts = '') {
+    void uploadFileToTransportRequestSOLMAN(
+        String changeId,
+        String transportRequestId,
+        String applicationId,
+        String filePath,
+        String endpoint,
+        String credentialsId,
+        String cmclientOpts = '') {
+
+        def args = [
+                '-cID', changeId,
+                '-tID', transportRequestId,
+                applicationId, "\"$filePath\""
+            ]
+
+        uploadFileToTransportRequest(
+            BackendType.SOLMAN,
+            '',
+            [],
+            endpoint,
+            credentialsId,
+            'upload-file-to-transport',
+            args,
+            cmclientOpts)
+    }
 
-        def args = null
+    void uploadFileToTransportRequestCTS(
+        String transportRequestId,
+        String applicationId,
+        String filePath,
+        String endpoint,
+        String credentialsId,
+        String cmclientOpts = '') {
+
+        def args = [
+                '-tID', transportRequestId,
+                "\"$filePath\""
+            ]
+
+        uploadFileToTransportRequest(
+            BackendType.CTS,
+            '',
+            [],
+            endpoint,
+            credentialsId,
+            'upload-file-to-transport',
+            args,
+            cmclientOpts)
+    }
 
-        if(type == BackendType.SOLMAN) {
-            args = ['-cID', changeId,
-                    '-tID', transportRequestId,
-                    applicationId, "\"$filePath\""]
-        } else if (type == BackendType.CTS) {
-            args = ['-tID', transportRequestId,
-                    "\"$filePath\""]
-        } else {
+    void uploadFileToTransportRequestRFC(
+        String dockerImage,
+        List dockerOptions,
+        String transportRequestId,
+        String applicationId,
+        String filePath,
+        String endpoint,
+        String credentialsId,
+        String developmentInstance,
+        String developmentClient,
+        String applicationDescription,
+        String abapPackage) {
+
+        def args = [
+                "--env ABAP_DEVELOPMENT_INSTANCE=${developmentInstance}",
+                "--env ABAP_DEVELOPMENT_CLIENT=${developmentClient}",
+                "--env ABAP_APPLICATION_NAME=${applicationId}",
+                "--env ABAP_APPLICATION_DESC=${applicationDescription}",
+                "--env ABAP_PACKAGE=${abapPackage}",
+                "--env ZIP_FILE_URL=${filePath}",
+                "--env TRANSPORT_DESCRIPTION=TODO" // under discussion, maybe better simply via description
```

Comment:
Reminder: How to get the description from the outside?

Reply:
> Reminder: How to

thanks for the reminder. For that reason there is currently the `TODO` in the code.

Reply:
Could the description be an additional parameter ?

Reply:
I would like to suggest to start the review on the current HEAD commit rather than having a closer look into old commits. I apologize for this monster commit, it is way too long on the way. This is not good, but sometimes it happens, especially when providing new more the less heavyweight features ...